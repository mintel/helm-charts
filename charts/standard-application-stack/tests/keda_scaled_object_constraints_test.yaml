suite: Test Keda ScaledObject Constraints
templates:
  - keda-scaled-object.yaml
release:
  namespace: test-namespace
tests:
  - it: Creates a ScaledObject with non-constrained settings
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.minReplicaCount: 5
      autoscaling.maxReplicaCount: 30
      autoscaling.cooldownPeriod: 500
      autoscaling.pollingInterval: 60
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.maxReplicaCount
          value: 30
      - equal:
          path: spec.minReplicaCount
          value: 5
      - equal:
          path: spec.pollingInterval
          value: 60
      - equal:
          path: spec.cooldownPeriod
          value: 500

  - it: Creates a ScaledObject with constrained settings
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.minReplicaCount: 0
      autoscaling.maxReplicaCount: 500
      autoscaling.cooldownPeriod: 5
      autoscaling.pollingInterval: 5
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.maxReplicaCount
          value: 30
      - equal:
          path: spec.minReplicaCount
          value: 1
      - equal:
          path: spec.pollingInterval
          value: 10
      - equal:
          path: spec.cooldownPeriod
          value: 60

  - it: Ensure minReplicaCount cannot be greater than 10
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.minReplicaCount: 15
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.minReplicaCount
          value: 2

  - it: Check idleReplicaCount is equal to 0 when enableZeroReplicas is set
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.enableZeroReplicas: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.idleReplicaCount
          value: 0

  - it: Check trigger vars automatically populated
    set:
      global.clusterEnv: qa
      global.clusterRegion: eu-west-2
      autoscaling.enabled: true
      autoscaling.triggers.custom:
        - type: aws-cloudwatch
          metadata:
            namespace: AWS/SQS
            dimensionName: QueueName
            dimensionValue: test-queue
            metricName: ApproximateNumberOfMessagesVisible
        - type: aws-sqs-queue
          metadata:
            queueURL: test-queue
            queueLength: 2
        - type: prometheus
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
          - type: aws-cloudwatch
            metadata:
              awsRegion: eu-west-2
              identityOwner: operator
              namespace: AWS/SQS
              dimensionName: QueueName
              dimensionValue: test-queue
              metricName: ApproximateNumberOfMessagesVisible
          - type: aws-sqs-queue
            metadata:
              awsRegion: eu-west-2
              identityOwner: operator
              queueURL: test-queue
              queueLength: "2"
          - type: prometheus
            metadata:
              serverAddress: http://prometheus-k8s.monitoring.svc:9090

  - it: Check trigger vars can be overriden
    set:
      global.clusterEnv: qa
      global.clusterRegion: eu-west-2
      autoscaling.enabled: true
      autoscaling.triggers.custom:
        - type: aws-cloudwatch
          metadata:
            awsRegion: us-east-2
            identityOwner: pod
            namespace: AWS/SQS
            dimensionName: QueueName
            dimensionValue: test-queue
            metricName: ApproximateNumberOfMessagesVisible
        - type: aws-sqs-queue
          metadata:
            awsRegion: us-east-2
            identityOwner: pod
            queueURL: test-queue
            queueLength: 2
        - type: prometheus
          metadata:
            serverAddress: http://alt-prometheus-k8s.monitoring.svc:9090
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
          - type: aws-cloudwatch
            metadata:
              awsRegion: us-east-2
              identityOwner: pod
              namespace: AWS/SQS
              dimensionName: QueueName
              dimensionValue: test-queue
              metricName: ApproximateNumberOfMessagesVisible
          - type: aws-sqs-queue
            metadata:
              awsRegion: us-east-2
              identityOwner: pod
              queueURL: test-queue
              queueLength: "2"
          - type: prometheus
            metadata:
              serverAddress: http://alt-prometheus-k8s.monitoring.svc:9090
