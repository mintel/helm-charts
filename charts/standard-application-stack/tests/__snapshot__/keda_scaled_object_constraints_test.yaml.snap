Check idleReplicaCount is equal to 0 when enableZeroReplicas is set:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      idleReplicaCount: 0
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
Check trigger vars automatically populated:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: eu-west-2
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            awsRegion: eu-west-2
            dimensionName: QueueName
            dimensionValue: test-queue
            identityOwner: operator
            metricName: ApproximateNumberOfMessagesVisible
            namespace: AWS/SQS
          type: aws-cloudwatch
        - metadata:
            awsRegion: eu-west-2
            identityOwner: operator
            queueLength: "2"
            queueURL: test-queue
          type: aws-sqs-queue
        - metadata:
            serverAddress: http://prometheus-k8s.monitoring.svc:9090
          type: prometheus
Check trigger vars can be overriden:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: eu-west-2
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            awsRegion: us-east-2
            dimensionName: QueueName
            dimensionValue: test-queue
            identityOwner: pod
            metricName: ApproximateNumberOfMessagesVisible
            namespace: AWS/SQS
          type: aws-cloudwatch
        - metadata:
            awsRegion: us-east-2
            identityOwner: pod
            queueLength: "2"
            queueURL: test-queue
          type: aws-sqs-queue
        - metadata:
            serverAddress: http://alt-prometheus-k8s.monitoring.svc:9090
          type: prometheus
Creates a ScaledObject with constrained settings:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 60
      fallback:
        failureThreshold: 3
        replicas: 1
      maxReplicaCount: 30
      minReplicaCount: 1
      pollingInterval: 10
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
Creates a ScaledObject with non-constrained settings:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 500
      fallback:
        failureThreshold: 3
        replicas: 5
      maxReplicaCount: 30
      minReplicaCount: 5
      pollingInterval: 60
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
Ensure minReplicaCount cannot be greater than 10:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
